Explorer/Assets/DCL/AvatarRendering/Wearables/Systems/ResolveWearableByPointerSystem.cs

```diff
@@ -268,15 +268,30 @@ private void SetDefaultWearables(bool defaultWearablesLoaded, IWearable wearable
 
             ReportHub.Log(GetReportCategory(), $"Request for wearable {wearable.GetHash()} failed, loading default wearable");
 
-            // This section assumes that the default wearables were successfully loaded.
-            // Waiting for the default wearable should be moved to the loading screen
+            var defaultWearable = wearableCatalog.GetDefaultWearable(bodyShape, wearable.GetCategory());
             if (wearable.IsUnisex())
             {
-                wearable.WearableAssetResults[BodyShape.MALE] = wearableCatalog.GetDefaultWearable(BodyShape.MALE, wearable.GetCategory()).WearableAssetResults[BodyShape.MALE];
-                wearable.WearableAssetResults[BodyShape.FEMALE] = wearableCatalog.GetDefaultWearable(BodyShape.FEMALE, wearable.GetCategory()).WearableAssetResults[BodyShape.FEMALE];
+                wearable.WearableAssetResults[BodyShape.MALE] = defaultWearable.WearableAssetResults[BodyShape.MALE];
+                wearable.WearableAssetResults[BodyShape.FEMALE] =
+                    defaultWearable.WearableAssetResults[BodyShape.FEMALE];
             }
             else
-                wearable.WearableAssetResults[bodyShape] = wearableCatalog.GetDefaultWearable(bodyShape, wearable.GetCategory()).WearableAssetResults[bodyShape];
+                wearable.WearableAssetResults[bodyShape] = defaultWearable.WearableAssetResults[bodyShape];
+
+            // If the default wearable is empty, we need to remove all the hiding/replacing/removing data
+            if (defaultWearable.GetUrn().Equals(WearablesConstants.EMPTY_DEFAULT_WEARABLE))
```

Comment:
It seems a bit strange that we need to manipulate DTO after we get Empty default wearable from cache

Reply:
Over here Im assuming that the wearable DTO has loaded correctly. Therefore we are just replacing the wearable asset bundle with the empty default one; not the whole wearable. So, thats the reason we have to modify the DTO of the wearable that has just failed. All other info is remained untouched and can be used (ie in the backpack)

Different scenario is that the wearable DTO has failed as well. If that is the case, then we can replace the whole wearable for the empty one; but I dont see this as a common scenario.

WDYT?

Reply:
Ok, thanks, now I see.
I propose the following changes then to improve encapsulation:

- Move intrusion to the DTO to an extension method of DTO itself
- Let's avoid `if (defaultWearable.GetUrn().Equals(WearablesConstants.EMPTY_DEFAULT_WEARABLE))`: we already know it from `wearableCatalog.GetDefaultWearable`, extend this method to return an additional flag (as `out` var for example)

Reply:
Done, but not completely sure that we need another static class for an extension method. I added a `Sanitize` method in `WearableDTO`, WDYT?

Reply:
It's absolutely fine.